lib/active_model_serializers/model.rb

```diff
@@ -1,44 +1,48 @@
 # ActiveModelSerializers::Model is a convenient
 # serializable class to inherit from when making
 # serializable non-activerecord objects.
+#
+# Caching support is by-default mixed in when subclassing.
+# To disable mixing in any caching support,
+# set ActiveModelSerializers::Model.add_caching_support = false
 module ActiveModelSerializers
   class Model
-    include ActiveModel::Model
     include ActiveModel::Serializers::JSON
+    include ActiveModel::Model
+    extend ActiveSupport::Autoload
+    autoload :Caching
+
+    class_attribute :add_caching_support
+    self.add_caching_support = true
+    class_attribute :attribute_names
+    self.attribute_names = []
 
     def self.attributes(*names)
-      attr_accessor(*names)
+      self.attribute_names |= names.map(&:to_sym)
+      # Silence redefinition of methods warnings
+      ActiveModelSerializers.silence_warnings do
```

Comment:
Why is this actually needed? Which method redefinition are you actually trying to silence?

Wouldn't

```ruby
names.map!(&to_sym)
attr_accessor(names - self.attribute_name)
self.attribute_names |= names
```

work instead of messing up with $VERBOSE?